File: couch/node/update/couch_update.js
#! /usr/bin/nodejs
// ---------------------------------------------------------------
//	couch_update.js
//
//						Nov/24/2014
// ---------------------------------------------------------------
var cradle = require ('cradle');
var text_manipulate = require('/var/www/data_base/common/node_common/text_manipulate');
// ---------------------------------------------------------------
console.log ("*** 開始 ***");

var key_in=process.argv[2];
var population_in=process.argv[3];

console.log (key_in + "\t" + population_in);

var name = 'nagano';
var cc = new (cradle.Connection);

var db = cc.database (name);

db.exists(function (err, exists)
	{
	if (err)
		{
		console.log('error', err);
		}
	else if (exists)
		{
		db.get (key_in,function (err,unit)
			{
			if (unit)
				{
				console.log (key_in);
				unit.population =  population_in;
				unit.date_mod = text_manipulate.get_current_date_proc ();
				console.log (unit);

				db.put (key_in,unit,function (err,res)
					{
					console.log (res);
					});
				}
			});
		console.log ("*** 終了 ***");
		}
	else
		{
		console.log('database does not exists.');
		}
	});

// ---------------------------------------------------------------

File: couch/perl/update/couch_update.pl
#! /usr/bin/perl
#
#	couch_update.pl
#
#					Jun/23/2014
# -------------------------------------------------------
use	strict;
use	warnings;
use	Encode;
use	JSON;
#
use lib '/var/www/data_base/common/perl_common';
use file_io;
use text_manipulate;
use uri_get;
#
# -------------------------------------------------------
print "*** 開始 ***\n";
#
my $key_in = $ARGV[0];
my $population_in = $ARGV[1];
print ("$key_in\t$population_in\n");
#
my $url = 'http://localhost:5984/nagano';
#
my $url_target = $url . "/" . $key_in;
my $str_json = uri_get::uri_get_proc ($url_target);
#
my $data = decode_json ($str_json);
#
print $str_json;
my %datax = %{$data};
#
if (exists $datax{"error"})
	{
	print "*** not exist ***\n";
	}
else
	{
	print "*** exists ***\n";
my $name = $data->{name};
$data->{population} = $population_in;
$data->{date_mod} = text_manipulate::get_date_mod_proc ();
#
print $key_in . "\t";
print encode ('utf-8',$name) . "\n";
#
#
my $json_str_new = encode_json ($data);
print $json_str_new . "\n";
#
uri_get::uri_put_proc ($url_target,$json_str_new);
	}
#
print "*** 終了 ***\n";
#
# -------------------------------------------------------
